quality_check/09-prior_information_matrix.Rmd

```diff
@@ -0,0 +1,225 @@
+# this is CC0
+downloadPathwayArchive(organism = "Danio rerio", format = "gmt",
+                       destpath = pathway_dir, date = "20190110")
+```
+
+Read the `.gmt` file in and convert to binary matrix
+
+```{r}
+zebrafish_wp_list <- 
+  GSA::GSA.read.gmt(file.path(pathway_dir, 
+                              "wikipathways-20190110-gmt-Danio_rerio.gmt"))
+```
+
+Clean up the list to make it more amenable to the binary prior information
+matrix we'll need for use with PLIER.
+
+```{r}
+names(zebrafish_wp_list$genesets) <- make.names(  # remove hyphens
+  gsub(" ", "_",  # convert spaces to underscores
+       gsub("%WikiPathways_20190110%", " ",  # remove date, WP information 
+            gsub("%Danio rerio", "",  # remove species information
+                 zebrafish_wp_list$geneset.names))
+  )
+)
+# remove pathway names and descriptions
+zebrafish_wp_list <- zebrafish_wp_list$genesets
+```
+
+We're going to use ZFIN identifiers instead of Entrez IDs, which are what are
+used in WikiPathways.
+
+```{r}
+# our goal is to get a binary matrix that represents gene-pathway relationships
+# to do this, we first need to melt the list into a data.frame
+zebrafish_wp_binary_df <- reshape2::melt(zebrafish_wp_list) %>%
+  # get sensible column names
+  dplyr::mutate(Gene = value, Pathway = L1) %>%
+  dplyr::select(Gene:Pathway) %>%
+  # conversion from Entrez IDs to ZFIN IDs with mapIds -- this should be 1:1
+  dplyr::mutate(ZFIN = mapIds(org.Dr.eg.db, 
+                              keys = as.character(Gene), 
+                              keytype = "ENTREZID", column = "ZFIN")) %>%
+  # drop Entrez IDs
+  dplyr::select(-Gene) %>%
+  # drop records missing ZFIN ids
+  dplyr::filter(!is.na(ZFIN)) %>%
+  # unique records only, just in case
+  dplyr::distinct() %>%
+  # get binary matrix, where the first column will be gene and the columns are
+  # pathways
+  reshape2::dcast(ZFIN ~ Pathway, fun.aggregate = length)
+```
+
+```{r}
+# now a more PLIER friendly format
+zebrafish_wp_binary_mat <- zebrafish_wp_binary_df %>%
+  tibble::column_to_rownames("ZFIN") %>%
+  as.matrix()
+saveRDS(zebrafish_wp_binary_mat, 
+        file.path(pathway_dir, "Danio_rerio_wikipathways_prior_info.RDS"))
+```
+
+## ZFIN
+
+Given the [licensing of the ZFIN resource](https://zfin.org/warranty.html) and 
+the fact that downloading via the web interface has the advantage of including 
+the archive date in the file name/header and including the column names, we've 
+obtained these files via the 
+[downloads webpage](https://zfin.org/downloads/archive/2019.01.25) using the
+date that is currently the most recent archive (`2019-01-25`).
+
+
+```{r}
+wildtype_expression_file <- 
+  file.path(pathway_dir, "wildtype-expression_fish_2019.01.25.txt")
+# we're only going to use the in situ data
+wildtype_expression_df <- readr::read_tsv(wildtype_expression_file, skip = 1) %>%
+  dplyr::filter(Assay == "mRNA in situ hybridization") %>%
+  dplyr::select(-X15)
```

Comment:
What's wrong with this column? Same as before?

Reply:
Both `.txt` files from ZFIN end up with an empty final column, yes.